flow/project.py

```diff
@@ -243,8 +243,22 @@ def __init__(self, name, job, cmd, directives=None, np=None):
         # script engine later.
         keys_set_by_user = set(directives.keys())
 
-        directives.setdefault(
-            'np', directives.get('nranks', 1) * directives.get('omp_num_threads', 1))
+        nranks = directives.get('nranks', 1)
+        omp_num_threads = directives.get('omp_num_threads', 1)
+        if callable(nranks):
+            if callable(omp_num_threads):
+                directives.setdefault(
+                    'np', lambda job: nranks(job)*omp_num_threads(job))
+            else:
+                directives.setdefault(
+                    'np', lambda job: nranks(job)*omp_num_threads)
+        elif callable(omp_num_threads):
+            directives.setdefault(
+                'np', lambda job: nranks*omp_num_threads(job))
+        else:
+            directives.setdefault(
+                'np', lambda job: nranks*omp_num_threads)
```

Comment:
Why does it have to be a callable in that case?

Reply:
It doesn't, I can change that. What are your thoughts regarding the options above? If you agree with Bradley I'll make that change as well, otherwise I'll just change this else clause to a non-callable case.

Reply:
@csadorf @mikemhenry thoughts?

Reply:
I'd prefer the alternative, because I don't like the amount of nested if-clauses here.

Reply:
Sounds good, I'll switch to what I pasted above. That version also fixes the callable issue.